cmd/agent/app/agent.go

```diff
@@ -226,7 +227,7 @@ func run(ctx context.Context, opts *options.Options) error {
 	return nil
 }
 
-func setupControllers(mgr controllerruntime.Manager, opts *options.Options, stopChan <-chan struct{}) error {
+func setupControllers(ctx context.Context, mgr controllerruntime.Manager, opts *options.Options, stopChan <-chan struct{}) error {
```

Comment:
`stopChan <-chan ` seems to be a redundant parameter.

Reply:
I have removed this parameter and use ctx.Done instead.  

There are still some controllers own the element `stopChan`, maybe we could change them to `ctx` later.